src/Common/src/CoreLib/System/MemoryExtensions.cs

```diff
@@ -1304,5 +1305,36 @@ public static bool Overlaps<T>(this ReadOnlySpan<T> first, ReadOnlySpan<T> secon
                 value, comparer);
             return BinarySearch(span, comparable);
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool IsTypeNumeric<T>(out int size)
+        {
+            if (typeof(T) == typeof(byte) || typeof(T) == typeof(sbyte))
+            {
+                size = sizeof(byte);
+                return true;
+            }
+
+            if (typeof(T) == typeof(char) || typeof(T) == typeof(short) || typeof(T) == typeof(ushort))
+            {
+                size = sizeof(char);
+                return true;
+            }
+
+            if (typeof(T) == typeof(int) || typeof(T) == typeof(uint))
+            {
+                size = sizeof(int);
+                return true;
+            }
+
+            if (typeof(T) == typeof(long) || typeof(T) == typeof(ulong))
+            {
+                size = sizeof(long);
+                return true;
+            }
+
+            size = 0;
```

Comment:
Nit: prefer `size = default;` here as the intent isn't that the size is zero, the intent is that the size is uninteresting.